packages/flutter/test/rendering/rendering_tester.dart

```diff
@@ -7,33 +7,93 @@ import 'package:flutter/gestures.dart';
 import 'package:flutter/rendering.dart';
 import 'package:flutter/scheduler.dart';
 import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart' show EnginePhase, fail;
 
-import 'package:flutter_test/flutter_test.dart' show EnginePhase;
+export 'package:flutter/foundation.dart' show FlutterError, FlutterErrorDetails;
 export 'package:flutter_test/flutter_test.dart' show EnginePhase;
 
 class TestRenderingFlutterBinding extends BindingBase with ServicesBinding, GestureBinding, SchedulerBinding, PaintingBinding, SemanticsBinding, RendererBinding {
+  /// Creates a binding for testing rendering library functionality.
+  ///
+  /// If [onErrors] is not null, it is called if [FlutterError] caught any errors
+  /// while drawing the frame. If [onErrors] is null and [FlutterError] caught at least
+  /// one error, this function fails the test. A test may override [onErrors] and
+  /// inspect errors using [takeFlutterErrorDetails].
+  TestRenderingFlutterBinding({ this.onErrors });
+
+  final List<FlutterErrorDetails> _errors = <FlutterErrorDetails>[];
+  VoidCallback onErrors;
```

Comment:
Please add doc to explain iterating removal of errors from _errors ie reason for Iterable sync*/yield

Reply:
Done.